src/Collection.php

```diff
@@ -89,17 +88,73 @@ protected function loop() {
 	/**
 	 * This function allow to format every item that is returned to the endpoint
 	 * the filter sends 3 params to the user so can be more easy to manipulate the
-	 * data based on certaim params.
+	 * data based on certain params.
 	 *
-	 * @param \WP_Post $the_post Current post object.
 	 * @return array The formated data from every item.
 	 */
-	protected function format_item( \WP_Post $the_post ) {
-		$item = [ 'id' => $the_post->ID ];
-		if ( $this->format_item ) {
-			$item = apply_filters( Filter::ITEM_FORMAT, $item, $the_post, $this->args );
+	protected function format_item() {
+		$the_post = get_post();
+
+		// Author.
+		$the_author = get_userdata( $the_post->post_author );
+
+		// Thumbnail.
+		if ( has_post_thumbnail( $the_post->ID ) ) {
+			$size = apply_filters( Filter::THUMBNAIL_SIZE, 'thumbnail', $the_post );
+
+			$attachment_id = get_post_thumbnail_id( $the_post->ID );
+
+			$src = wp_get_attachment_image_src( $attachment_id, $size );
+
+			$thumbnail = [
+				'src' 		=> $src[0],
+				'width'		=> $src[1],
+				'height'	=> $src[2],
+				'alt'		=> get_post_meta( $attachment_id, '_wp_attachment_image_alt', true ),
+			];
+		} else {
+			$thumbnail = false;
 		}
-		return $item;
+
+		// Taxonomies.
+		$taxonomies = get_object_taxonomies( $the_post );
+
+		$post_terms = [];
+
+		foreach ( $taxonomies as $taxonomy ) {
+			$post_terms[ $taxonomy ] = [];
+
+			$terms = wp_get_post_terms( $the_post->ID, $taxonomy );
+
+			foreach ( $terms as $term ) {
+				$post_terms[ $taxonomy ][] = [
+					'id' => $term->term_id,
+					'name' => $term->name,
+					'slug' => $term->slug,
+					'description' => $term->description,
+				];
+			}
+		}
+
+		// Put it all together.
+		$item = [
+			'id' => $the_post->ID,
+			'title' => $the_post->post_title,
+			'link' => get_permalink( $the_post->ID ),
+			'slug' => $the_post->post_name,
+			'excerpt' => get_the_excerpt(),
+			'author' => [
+				'id' => $the_author->ID,
+				'first_name' => $the_author->first_name,
+				'last_name' => $the_author->last_name,
+				'posts_link' => str_replace( site_url(), '', get_author_posts_url( $the_author->ID ) ),
```

Comment:
Well nothing major to crate an instance variable, after that does not require any extra step. Well if are small things we can improve I think is better small improvements made step by step instead of huge refactors. 

> On Apr 4, 2016, at 8:06 PM, Adam Fenton <notifications@github.com> wrote:
> 
> In src/Collection.php <https://github.com/moxie-leean/wp-endpoints-collection/pull/8#discussion_r58473918>:
> 
> > +				];
> > +			}
> > +		}
> > +
> > +		// Put it all together.
> > +		$item = [
> > +			'id' => $the_post->ID,
> > +			'title' => $the_post->post_title,
> > +			'link' => get_permalink( $the_post->ID ),
> > +			'slug' => $the_post->post_name,
> > +			'excerpt' => get_the_excerpt(),
> > +			'author' => [
> > +				'id' => $the_author->ID,
> > +				'first_name' => $the_author->first_name,
> > +				'last_name' => $the_author->last_name,
> > +				'posts_link' => str_replace( site_url(), '', get_author_posts_url( $the_author->ID ) ),
> To call it once we'd have to pass it into the function :-( I'm pretty sure wp caches this stuff so there's no need to worry about performance
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/moxie-leean/wp-endpoints-collection/pull/8/files/e39f642d6dc877ed26ee7a3f2b4cc81077ed77ec#r58473918>